src/main.cpp

```diff
@@ -117,8 +119,11 @@ int encode(Image &image, const std::array<std::uint8_t, 32> &password, const std
 
     // Generate the Salt and IV
     std::uint8_t salt[16], iv[16];
-    random.get(salt, sizeof(salt));
-    random.get(iv,   sizeof(iv));
+	if (!random.get(salt, sizeof salt) || !random.get(iv, sizeof iv))
+	{
+		std::cout << "ERROR: Unable to generate random number\n";
```

Comment:
```suggestion
		std::cerr << "ERROR: Unable to generate random number" << std::endl;
```
Please output errors to standard error, and use std::endl to make sure that the buffer is flushed.